src/Maestro/tests/Maestro.ScenarioTests/MaestroScenarioTestBase.cs

```diff
@@ -153,14 +181,50 @@ public async Task DeleteDefaultTestChannelAsync(string testChannelName, string r
             await RunDarcAsync("delete-default-channel", "--channel", testChannelName, "--repo", repoUri, "--branch", branch).ConfigureAwait(false);
         }
 
-        public async Task<AsyncDisposableValue<string>> CreateSubscriptionAsync(string sourceChannelName, string sourceRepo, string targetRepo, string targetBranch, string updateFrequency)
+        public async Task<AsyncDisposableValue<string>> CreateSubscriptionAsync(string sourceChannelName, string sourceRepo, string targetRepo, string targetBranch, string updateFrequency, string sourceOrg = "dotnet",
+            List<string> additionalOptions = null, bool sourceIsAzDo = false, bool targetIsAzDo = false)
         {
-            string output = await RunDarcAsync("add-subscription", "-q", "--no-trigger",
+            string sourceUrl = sourceIsAzDo ? GetAzDoRepoUrl(sourceRepo, azdoProject: sourceOrg) : GetRepoUrl(sourceOrg, sourceRepo);
+            string targetUrl = targetIsAzDo ? GetAzDoRepoUrl(targetRepo) : GetRepoUrl(targetRepo);
+
+            string[] command = {"add-subscription", "-q", "--no-trigger",
                 "--channel", sourceChannelName,
-                "--source-repo", GetRepoUrl("dotnet", sourceRepo),
-                "--target-repo", GetRepoUrl(targetRepo),
+                "--source-repo", sourceUrl,
+                "--target-repo", targetUrl,
                 "--target-branch", targetBranch,
-                "--update-frequency", updateFrequency).ConfigureAwait(false);
+                "--update-frequency", updateFrequency};
+
+            if (additionalOptions != null)
+            {
+                command = command.Concat(additionalOptions).ToArray();
+            }
+
+            string output = await RunDarcAsync(command).ConfigureAwait(false);
+
+            Match match = Regex.Match(output, "Successfully created new subscription with id '([a-f0-9-]+)'");
+            if (match.Success)
+            {
+                string subscriptionId = match.Groups[1].Value;
+                return AsyncDisposableValue.Create(subscriptionId, async () =>
+                {
+                    TestContext.WriteLine($"Cleaning up Test Subscription {subscriptionId}");
+                    try
+                    {
+                        await RunDarcAsync("delete-subscriptions", "--id", subscriptionId, "--quiet").ConfigureAwait(false);
+                    }
+                    catch (MaestroTestException)
+                    {
+                        // If this throws an exception the most likely cause is that the subscription was deleted as part of the test case
+                    }
+                });
+            }
+
+            throw new MaestroTestException("Unable to create subscription.");
```

Comment:
Wait... this method _always_ throws.  There are no returns in this method, and this "throw" isn't in any sort of statement.

Reply:
The return is on line 208

Reply:
Oh wow... I didn't notice that.

Can we swap the order... so if it says `if (!match.Succes) { throw }`?

Technically just because that string isn't there, doesn't mean we don't need to clean up the subscription.  I'm not sure how to handle that without the tests leaking subscriptions all over the place.